config/scc/scc.yaml

```diff
@@ -0,0 +1,30 @@
+apiVersion: security.openshift.io/v1
+kind: SecurityContextConstraints
+metadata:
+  name: test-scc
+allowHostDirVolumePlugin: false
+allowHostIPC: false
+allowHostNetwork: false
+allowHostPID: false
+allowHostPorts: false
+allowPrivilegeEscalation: true
+allowPrivilegedContainer: false
+allowedCapabilities: [AUDIT_WRITE]
+fsGroup:
+  type: RunAsAny
+priority: 11
```

Comment:
When I look at the standard SCCs, only `anyuid` has a non-nil priority (its value is 10); so this is creating an SCC with a priority higher than anything built-in.
My intuition from looking at [the sorting](https://github.com/openshift/origin/blob/99ad04e039e991900ba73a7ec9f03459af704a5f/vendor/github.com/openshift/apiserver-library-go/pkg/securitycontextconstraints/util/sort/bypriority.go#L22-L25) is that we would want something less than `anyuid`.
It also looks like a nil priority would lead to [sorting by "restricted-ness"](https://github.com/openshift/origin/blob/99ad04e039e991900ba73a7ec9f03459af704a5f/vendor/github.com/openshift/apiserver-library-go/pkg/securitycontextconstraints/util/sort/bypriority.go#L31-L39) to break the tie with the other built-ins. Since we are less restrictive, it may get skipped in favor of `restricted`.

What I'm saying is... I don't know what this priority should be, but my guess would be 0 < pri < 10

Reply:
Got it. Will set to 1 for now and we can try it out.

Reply:
@JohnStrunk Actually perhaps the priority sorting concerns here are not real - 
I tested that when a pod specifies `runAsUser: 0` in its spec.securityContext then `restricted` will be rejected and skipped by this code below and will then try to match with the rest of the SCC's.

https://github.com/openshift/apiserver-library-go/blob/release-4.6/pkg/securitycontextconstraints/sccadmission/admission.go#L193-L231

I just tested it with nil priority and that works - will update the sample yamls in the PR just for the example.

So as long as we set the mover RBAC to allow "use" of a custom provided Scribe SCC, and specify the requirements in the pod spec.securityContext, we should be OK even with nil priority, which seems to be the right approach here based on the fact that all other builtin SCC's other than anyuid use it.

We can also set to priority to 1 but I suspect that it might create the same problems we had as with 11 which is to override builtin SCC's with nil priority such as hostaccess or privileged...

WDYT?

Reply:
Thanks for digging into this. Nil sounds like a good place to start.